WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -187,8 +193,11 @@ final class EditableOrderViewModel: ObservableObject {
             onVariationSelected: { [weak self] variation, parentProduct in
                 guard let self = self else { return }
                 self.addProductVariationToOrder(variation, parent: parentProduct)
+            }, onMultipleSelectionCompleted: { [weak self] _ in
+                guard let self = self else { return }
+                self.syncOrderItems(products: self.selectedProducts, variations: self.selectedProductVariations)
```

Comment:
As covered in pairing, we could really simplify the productSelector and productVariationSelector by redefining their contract.

Ideally, they could take an input (the currently selected items) and give an output (a single call to `onMultipleSelectionCompleted` which passes out the full details of what's selected when the user taps the button.

Currently, we rely on the other state which it's emitted over its lifecycle via `onProductSelected` and `onProductVariationSelected`.

Simplifying this will reduce the number of individual bugs we need to find and fix. I think it's outside the scope of this PR, but there are a number of tricky to reproduce issues in this PR which we found together, and squashing each one of them would be more effort than making the interface better.

Caveat – we've not checked what this change would mean for coupons. It shouldn't bee allowed to explode in to a huge refactor – if it's going that way, we should check this decision quickly.

Reply:
Thanks for your time and the walkthrough through changing the interface, that was definitely helpful. I'll keep digging into what changes are required and how these affect coupons today, so we have a better idea if there's any blocker that was unaccounted for. 

Something that I've seen so far that will need some refactor is when we "toggle all variations", which we don't do in products but coupons does: Coupons can easily access the variations IDs from the product model object via `product.variations` property, but can't get the variation models themselves, which we'll be needed when we changing the interface. Most likely we can fetch from the remote on a first call and then rely on local storage for this, I'll take a further look today.